scipy/optimize/tests/test_optimize.py

```diff
@@ -2239,6 +2241,198 @@ def test_gh12696():
+        warnings.simplefilter('ignore', category=UserWarning)
+
+        # the minimization with one parameter having lb == ub
+        res = optimize.minimize(
+            wrapped_rosen, x0, method=method, bounds=bounds, callback=callback
+        )
+        assert res.success
+        assert_allclose(res.x, np.r_[best_x.x, 2.0], rtol=3e-6)
+        if hasattr(res, "jac") and method in {"Powell", "L-BFGS-B", "SLSQP"}:
+            assert np.array(res.jac).size == 2
+            assert np.isnan(res.jac[1])
+
+        # check that SLSQP can deal with equal bounds if it has constraints
+        if method == "slsqp":
+            constr_best_x = optimize.minimize_scalar(
+                f, method="bounded", bounds=(2.0, 3.0)
+            )
+
+            def con(x):
+                assert len(x) == 2
+                return x[0]
+
+            def conjac(x):
+                assert len(x) == 2
+                return 1
+
+            constr = optimize.NonlinearConstraint(con, 2, 3)
+            res = optimize.minimize(
+                optimize.rosen, x0, method=method, bounds=bounds,
+                constraints=[constr]
+            )
+            assert res.success
+            assert_allclose(res.x, np.r_[constr_best_x.x, 2.0], rtol=3e-6)
+
+            constr = optimize.NonlinearConstraint(con, 2, 3, jac=conjac)
+            res = optimize.minimize(
+                optimize.rosen, x0, method=method, bounds=bounds,
+                constraints=[constr]
+            )
+            assert res.success
+            assert_allclose(res.x, np.r_[constr_best_x.x, 2.0], rtol=3e-6)
+
+        # check that minimize works if func returns func and grad.
+        if method not in ['Powell']:
+            # powell doesn't use jac
+            res = optimize.minimize(
+                func_and_grad, x0, method=method, bounds=bounds, jac=True
+            )
+            assert res.success
+            assert_allclose(res.x, np.r_[best_x.x, 2.0], rtol=3e-6)
+
+# --- Test minimize with equal upper and lower bounds --- #
+
+np.random.seed(0)
+eb_x0 = np.random.rand(4)
+eb_lb = np.array([0, 2, -1, -1])
+eb_ub = np.array([3, 2, 2, -1])
+eb_i = (eb_lb == eb_ub)
+
+
+def eb_rosen(x):
+    # assert x.size == 4
+    # assert_allclose(x[eb_i], eb_lb[eb_i])
+    return optimize.rosen(x)
+
+
+def eb_drosen(x):
+    # assert x.size == 4
+    # assert_allclose(x[eb_i], eb_lb[eb_i])
+    return optimize.rosen_der(x)
+
+
+def eb_callback(x, *args):
+    assert x.size == 4
+    assert_allclose(x[eb_i], eb_lb[eb_i])
+
+
+# test without gradient, with gradient, and with combined objective/gradient
+eb_kwds = ({"fun": eb_rosen, "jac": False},
+           {"fun": eb_rosen, "jac": eb_drosen},
+           {"fun": (lambda x: (eb_rosen(x), eb_drosen(x))),
+            "jac": True})
+
+# test with both old- and new-style bounds
+eb_bound_types = (lambda lb, ub: list(zip(lb, ub)), Bounds)
+
+
+# constraints and jacobians
+def eb_constraint1(x):
+    return x[0:1] - 1
```

Comment:
Should add the checks on `x` here and below.

Reply:
Can't, at least not without modifications. See [here](https://github.com/scipy/scipy/issues/14565#issuecomment-894995307).